### DISCUSSION TITLE
[Chapter 4] 코드 외부와 데이터 교환은 국소화하기 (57 page)

### DISCUSSION THREAD

**Discussion by firewoody237** [cowaine/is-my-code-that-strange#23] [Chapter 04-05 (불변, 응집도)] [answered]
이 부분은 어떤 주의점으로 저자가 적은건지 잘 이해가 안갑니다.

A Request에서 특정 Row를 조회 해 왔는데, 그 사이에 B Request에서 해당 Row를 변경하는.. 소위 Transaction으로 묶이지 않는 상태를 설명하려고 하는 걸까요?

**Comment by Irisation23** [comment, marked as answer]
불변에 대해 설명을 하는 중간에 해당 내용을 저자가 넣었고, `국소화(localization)` 라는 키워드에 주안점을 맞춰 본다면 @firewoody237 님의 예처럼 데이터 교환이 일어날 경우에는 불변(데이터의 삭제 등등)이 지켜지지 않을 수 있으니 주의하라 정도로 해석해 볼 수 있지 않을까요?

구체적으로 불변과 4.3.4 코드 외부와 데이터 교환은 국소화하기는 어떤 부분이 맞닿아 있는지 크게 와닿지는 않네요..

**Comment by coalong** [comment]
저자가 리포지토리 패턴(repository pattern) 언급하고 있는 것으로 이해했을 때,
비지니스 로직과 외부 데이터(ex. DB, File)를 분리해야 된다는 것을 주의하는 것 같습니다.

**Comment by Irisation23** [comment]
### 체크 포인트

- 해당 파트는 책의 설명이 모호함 (e.g 왜 `불변` 파트에 데이터 교환의 국소화가 나왔는지?)
- 크게 본다면 트랜잭션과 헥사고날 아키텍쳐와 관련이 있을 수 있음
  - `Persitence Adapter` 패턴
